test/main.sh

```diff
@@ -218,6 +219,12 @@ run_test() {
   fi
 
   END_TIME=$(date +%s)
+  cd "${cwd}"
+
+  # Run the panic checker after every test. This ensures that the test fails if any panics occur (as long as DEBUG is set).
+  # It is possible for a suite to succeed with panics in cases where a command is expected to fail (e.g. `! <cmd> || false`)
+  # but panics are never acceptable at runtime.
+  panic_checker "${TEST_DIR}"
```

Comment:
TEST_DIR is a directory but the script expects a file, no?

Reply:
`panic_checker` is a test util which lists all LXD daemon directories and performs the check on all of them. There are definitely some things to improve here.